integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/KafkaMicroBatchStreamStrategy.java

```diff
@@ -0,0 +1,115 @@
+import io.openlineage.spark.api.DatasetFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.spark.sql.connector.read.streaming.Offset;
+import org.apache.spark.sql.execution.datasources.v2.StreamingDataSourceV2Relation;
+import org.apache.spark.sql.types.StructType;
+import scala.Option;
+
+@Slf4j
+final class KafkaMicroBatchStreamStrategy extends StreamStrategy {
+  private static final String KAFKA_SOURCE_OFFSET_CLASS_NAME =
+      "org.apache.spark.sql.kafka010.KafkaSourceOffset";
+
+  public KafkaMicroBatchStreamStrategy(
+      DatasetFactory<InputDataset> inputDatasetDatasetFactory,
+      StreamingDataSourceV2Relation relation) {
+    super(inputDatasetDatasetFactory, relation);
+  }
+
+  @Override
+  public List<InputDataset> getInputDatasets() {
+    Optional<String> bootstrapServersOpt = getBootstrapServers();
+    Set<String> topics = getTopics();
+    if (bootstrapServersOpt.isPresent() && !topics.isEmpty()) {
+      return generateInputDatasets(bootstrapServersOpt.get(), topics);
+    } else {
+      log.warn(
+          "Could not generate an input dataset because bootstrapServers need to be present and at least one topic must exist");
+      return Collections.emptyList();
+    }
+  }
+
+  private Optional<String> getBootstrapServers() {
+    Optional<Map<String, Object>> executorKafkaParams =
+        tryReadField(relation.stream(), "executorKafkaParams");
+    return executorKafkaParams.map(m -> (String) m.get("bootstrap.servers"));
+  }
+
+  private Set<String> getTopics() {
+    Option<Offset> offsetOption = relation.startOffset();
+    if (offsetOption.isDefined()) {
+      Offset offset = offsetOption.get();
+      Class<?> offetClass = offset.getClass();
+      String offsetClassName = offetClass.getCanonicalName();
+      if (KAFKA_SOURCE_OFFSET_CLASS_NAME.equals(offsetClassName)) {
+        Optional<scala.collection.immutable.Map<TopicPartition, Long>> topicPartitionsMap =
+            tryReadField(offset, "partitionToOffsets");
+        return topicPartitionsMap
+            .map(ScalaConversionUtils::toJavaMap)
+            .map(Map::keySet)
+            .map(this::convertTopicPartitions)
+            .orElseGet(Collections::emptySet);
+      } else {
+        log.warn("Encountered an unsupported offset class: {}", offsetClassName);
+        return Collections.emptySet();
+      }
+    } else {
+      return Collections.emptySet();
+    }
+  }
+
+  private List<InputDataset> generateInputDatasets(
+      String bootstrapServersConfig, Collection<String> topics) {
+    String[] bootstrapServers = bootstrapServersConfig.split(",");
+    String namespace = "kafka://" + bootstrapServers[0];
```

Comment:
Isn't this different behaviour than you defined for output within this PR? 

General issue for this problem is: https://github.com/OpenLineage/OpenLineage/issues/560